src/devices/Bmxx80/Bme280.cs

```diff
@@ -0,0 +1,122 @@
+using Iot.Device.Bmxx80.Register;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents a BME280 temperature, barometric pressure and humidity sensor.
+    /// </summary>
+    public class Bme280 : Bmx280Base
+    {
+        /// <summary>
+        /// The expected chip ID of the BME280.
+        /// </summary>
+        private const byte DeviceId = 0x60;
+
+        /// <summary>
+        /// Calibration data for the <see cref="Bme680"/>.
+        /// </summary>
+        private readonly Bme280CalibrationData _bme280Calibration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bme280"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Bme280(I2cDevice i2cDevice)
+            : base(DeviceId, i2cDevice)
+        {
+            var bme280CalibrationData = new Bme280CalibrationData();
+            bme280CalibrationData.ReadFromDevice(this);
+            _bme280Calibration = bme280CalibrationData;
+            _calibrationData = bme280CalibrationData;
+
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        /// <summary>
+        /// Get the current sample rate for humidity measurements.
+        /// </summary>
+        /// <returns>The humidity <see cref="Sampling"/>.</returns>
+        public Sampling ReadHumiditySampling()
+        {
+            byte status = Read8BitsFromRegister((byte)Bme280Register.CTRL_HUM);
+            status = (byte)(status & 0b_0000_0111);
+            return ByteToSampling(status);
+        }
+
+        /// <summary>
+        /// Sets the humidity sampling to the given value.
+        /// </summary>
+        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
+        public void SetHumiditySampling(Sampling sampling)
+        {
+            byte status = Read8BitsFromRegister((byte)Bme280Register.CTRL_HUM);
+            status = (byte)(status & 0b_1111_1000);
+            status = (byte)(status | (byte)sampling);
+            _i2cDevice.Write(new[] { (byte)Bme280Register.CTRL_HUM, status });
+
+            // Changes to the above register only become effective after a write operation to "CTRL_MEAS".
+            byte measureState = Read8BitsFromRegister((byte)Bmx280Register.CTRL_MEAS);
+            _i2cDevice.Write(new[] { (byte)Bmx280Register.CTRL_MEAS, measureState });
+        }
+
+        /// <summary>
+        /// Reads the Humidity from the sensor as %rH.
+        /// </summary>
+        /// <returns>Returns a percentage from 0 to 100.</returns>
+        public async Task<double> ReadHumidityAsync()
+        {
+            if (ReadPowerMode() == Bmx280PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadHumiditySampling()));
+            }
+
+            // Read the temperature first to load the t_fine value for compensation.
+            if (TemperatureFine == int.MinValue)
```

Comment:
Same comment about TemperatureFine

Reply:
Resolved by Bmx280base's constructor which this inherits from